ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/BodyValidation.java

```diff
@@ -89,16 +88,23 @@ public static Hash withdrawalsRoot(final List<Withdrawal> withdrawals) {
   }
 
   /**
-   * Generates the requests root for a list of requests
+   * Generates the requests hash for a list of requests
    *
    * @param requests list of request
    * @return the requests root
    */
-  public static Hash requestsRoot(final List<Request> requests) {
-    final MerkleTrie<Bytes, Bytes> trie = trie();
-    IntStream.range(0, requests.size())
-        .forEach(i -> trie.put(indexKey(i), RequestEncoder.encodeOpaqueBytes(requests.get(i))));
-    return Hash.wrap(trie.getRootHash());
+  public static Hash requestsHash(final List<Request> requests) {
```

Comment:
It is important to remember that if a specific request type is absent, we must still include its type in the hash calculation.

For example, if there are no withdrawal requests, the hash will be:
```
hash = keccac256(keccac256(0x00 ++ <encoded_deposits>) ++ keccac256(0x01) ++ keccac256(0x02 ++ <encoded_consolidations>))
```

Reply:
Yep, found that out running the execution spec tests. I'm handling that by ensuring the output from processing requests includes empty bytes if there is no data from processing the request i.e. they are absent. So when doing the hash calculation we will always have 3 requests but they might have no bytes for the request data. Doing it at that level so it is handled by the hash calculation and when sending requests via the engine API.